### ISSUE TITLE
Crashes on linux almost immediately while starting a new game

### ISSUE THREAD

**Issue by CReimer** [Try/OpenGothic#509]
Crashes right after selecting "New Game". The loading bar reaches about 75% and boom

Gothic 1:
```
christopher@adanos ~ % Gothic2Notr -g /home/christopher/Gothic
OpenGothic v1.0 dev
no *.ini file in path - using default settings
GPU = AMD Radeon RX 7900 XT (RADV GFX1100)
Depth format = Depth32F Shadow format = Depth16
[phoenix] world: parsing object [MeshAndBsp % 0 0]
[phoenix] bsp_tree: parsing chunk c000
[phoenix] bsp_tree: parsing chunk c010
[phoenix] bsp_tree: parsing chunk c040
[phoenix] bsp_tree: parsing chunk c045
[phoenix] bsp_tree: parsing chunk c050
[phoenix] bsp_tree: parsing chunk c0ff
[phoenix] world: parsing object [VobTree % 0 0]
[phoenix] world: parsing object [WayNet % 0 0]
[phoenix] world: parsing object [EndMarker % 0 0]
[phoenix] model_script: detected invalid use of keyword aniin "Model" block. Ignoring rest of script.
unable to load sound fx: WOOD_NIGHT2
[phoenix] model script: syntax error (line 3186, column 2): comments must start with two slashes
comb not found: t_FIREPLACE_Stand_2_S0 -> c_FP_STAND_2_S0_(c_FP_STAND_2_S0_2)
comb not found: t_FIREPLACE_S0_2_Stand -> c_FP_STAND_2_S0_(c_FP_STAND_2_S0_2)
alias not found: t_Walk_2_WalkBL -> t_walkL_2_Walk
/usr/include/c++/13.2.1/string_view:258: constexpr const std::basic_string_view<_CharT, _Traits>::value_type& std::basic_string_view<_CharT, _Traits>::operator[](size_type) const [with _CharT = char; _Traits = std::char_traits<char>; const_reference = const char&; size_type = long unsigned int]: Assertion '__pos < this->_M_len' failed.

---crashlog(SIGABRT)---
GPU: AMD Radeon RX 7900 XT (RADV GFX1100)
#1:  - /usr/lib/libc.so.6(+0x8e83c) [0x7fb70908e83c]
#2: raise - /usr/lib/libc.so.6(raise+0x18) [0x7fb70903e668]
#3: abort - /usr/lib/libc.so.6(abort+0xd7) [0x7fb7090264b8]
#4: abort - /usr/lib/libstdc++.so.6(+0xdd3b2) [0x7fb7094dd3b2]
#5: WayPoint::checkName(std::basic_string_view<char, std::char_traits<char> >) const - Gothic2Notr(_ZNK8WayPoint9checkNameESt17basic_string_viewIcSt11char_traitsIcEE+0x9a) [0x561b6d41045a]
#6: WayMatrix::findPoint(std::basic_string_view<char, std::char_traits<char> >, bool) const - Gothic2Notr(_ZNK9WayMatrix9findPointESt17basic_string_viewIcSt11char_traitsIcEEb+0xb3) [0x561b6d40ae83]
#7: GameScript::ta_min(std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN10GameScript6ta_minESt10shared_ptrIN7phoenix5c_npcEEiiiiiSt17basic_string_viewIcSt11char_traitsIcEE+0x67) [0x561b6d2259e7]
#8: std::_Function_handler<void (std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >), GameScript::bindExternal<void (GameScript::*)(std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >)>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void (GameScript::*)(std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >))::{lambda((auto:1)...)#1}>::_M_invoke(std::_Any_data const&, std::shared_ptr<phoenix::c_npc>&&, int&&, int&&, int&&, int&&, int&&, std::basic_string_view<char, std::char_traits<char> >&&) - Gothic2Notr(_ZNSt17_Function_handlerIFvSt10shared_ptrIN7phoenix5c_npcEEiiiiiSt17basic_string_viewIcSt11char_traitsIcEEEZN10GameScript12bindExternalIMS9_FvS3_iiiiiS7_EEEvRKNSt7__cxx1112basic_stringIcS6_SaIcEEET_EUlDpT_E_E9_M_invokeERKSt9_Any_dataOS3_OiSS_SS_SS_SS_OS7_+0x90) [0x561b6d24ec70]
#9: std::_Function_handler<void (phoenix::vm&), phoenix::vm::register_external<void, std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> > >(std::basic_string_view<char, std::char_traits<char> >, std::function<void (std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >)> const&)::{lambda(phoenix::vm&)#1}>::_M_invoke(std::_Any_data const&, phoenix::vm&) - Gothic2Notr(_ZNSt17_Function_handlerIFvRN7phoenix2vmEEZNS1_17register_externalIvJSt10shared_ptrINS0_5c_npcEEiiiiiSt17basic_string_viewIcSt11char_traitsIcEEEEEvSB_RKSt8functionIFT_DpT0_EEEUlS2_E_E9_M_invokeERKSt9_Any_dataS2_+0x12c) [0x561b6d25bb8c]
#10: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0x966) [0x561b6d496846]
#11: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x561b6d496e61]
#12: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0xb28) [0x561b6d496a08]
#13: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x561b6d496e61]
#14: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(+0x2379fa) [0x561b6d2379fa]
#15: GameScript::initializeInstanceNpc(std::shared_ptr<phoenix::c_npc> const&, unsigned long) - Gothic2Notr(_ZN10GameScript21initializeInstanceNpcERKSt10shared_ptrIN7phoenix5c_npcEEm+0x3a4) [0x561b6d238444]
#16: Npc::Npc(World&, unsigned long, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN3NpcC1ER5WorldmSt17basic_string_viewIcSt11char_traitsIcEE+0x4e0) [0x561b6d3eeaf0]
#17: WorldObjects::addNpc(unsigned long, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN12WorldObjects6addNpcEmSt17basic_string_viewIcSt11char_traitsIcEE+0x75) [0x561b6d41f645]
#18: GameScript::wld_insertnpc(int, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN10GameScript13wld_insertnpcEiSt17basic_string_viewIcSt11char_traitsIcEE+0x1c) [0x561b6d22833c]
#19: std::_Function_handler<void (phoenix::vm&), phoenix::vm::register_external<void, int, std::basic_string_view<char, std::char_traits<char> > >(std::basic_string_view<char, std::char_traits<char> >, std::function<void (int, std::basic_string_view<char, std::char_traits<char> >)> const&)::{lambda(phoenix::vm&)#1}>::_M_invoke(std::_Any_data const&, phoenix::vm&) - Gothic2Notr(_ZNSt17_Function_handlerIFvRN7phoenix2vmEEZNS1_17register_externalIvJiSt17basic_string_viewIcSt11char_traitsIcEEEEEvS8_RKSt8functionIFT_DpT0_EEEUlS2_E_E9_M_invokeERKSt9_Any_dataS2_+0x62) [0x561b6d249ef2]
#20: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0x966) [0x561b6d496846]
#21: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x561b6d496e61]
#22: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0xb28) [0x561b6d496a08]
#23: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x561b6d496e61]
#24: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(+0x2897aa) [0x561b6d2897aa]
#25: GameSession::initScripts(bool) - Gothic2Notr(_ZN11GameSession11initScriptsEb+0x52b) [0x561b6d289d2b]
#26: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - Gothic2Notr(_ZN11GameSessionC1ENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x239) [0x561b6d289fb9]
#27: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - Gothic2Notr(+0x36b1ac) [0x561b6d36b1ac]
#28: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - Gothic2Notr(+0x2afa36) [0x561b6d2afa36]
#29: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - /usr/lib/libstdc++.so.6(+0xe1943) [0x7fb7094e1943]
#30: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - /usr/lib/libc.so.6(+0x8c9eb) [0x7fb70908c9eb]
#31: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - /usr/lib/libc.so.6(+0x110dfc) [0x7fb709110dfc]

[1]    13458 IOT instruction (core dumped)  Gothic2Notr -g /home/christopher/Gothic
```

Gothic 2:
```
christopher@adanos ~ % Gothic2Notr -g /home/christopher/Gothic\ II
OpenGothic v1.0 dev
no *.ini file in path - using default settings
GPU = AMD Radeon RX 7900 XT (RADV GFX1100)
Depth format = Depth32F Shadow format = Depth16
[phoenix] world: parsing object [MeshAndBsp % 0 0]
[phoenix] bsp_tree: parsing chunk c000
[phoenix] bsp_tree: parsing chunk c010
[phoenix] bsp_tree: parsing chunk c040
[phoenix] bsp_tree: parsing chunk c045
[phoenix] bsp_tree: parsing chunk c050
[phoenix] bsp_tree: parsing chunk c0ff
[phoenix] mesh: 1 bytes remaining in section b020
[phoenix] world: parsing object [VobTree % 0 0]
[phoenix] world: parsing object [WayNet % 0 0]
[phoenix] world: parsing object [EndMarker % 0 0]
invalid particle system: "INVISIBLE_VOBBOX.3DS"
unable to load sound fx: ENV_NIGHT_TONSOFINSECTS
unable to load sound fx: OW_BIRD11
unable to load mesh "_1HST1.MDS"
unable to load mesh "_2HST1.MDS"
unable to load mesh "_BOWT1.MDS"
unable to load mesh "_CBOWT1.MDS"
/usr/include/c++/13.2.1/string_view:258: constexpr const std::basic_string_view<_CharT, _Traits>::value_type& std::basic_string_view<_CharT, _Traits>::operator[](size_type) const [with _CharT = char; _Traits = std::char_traits<char>; const_reference = const char&; size_type = long unsigned int]: Assertion '__pos < this->_M_len' failed.

---crashlog(SIGABRT)---
GPU: AMD Radeon RX 7900 XT (RADV GFX1100)
#1:  - /usr/lib/libc.so.6(+0x8e83c) [0x7f436308e83c]
#2: raise - /usr/lib/libc.so.6(raise+0x18) [0x7f436303e668]
#3: abort - /usr/lib/libc.so.6(abort+0xd7) [0x7f43630264b8]
#4: abort - /usr/lib/libstdc++.so.6(+0xdd3b2) [0x7f43634dd3b2]
#5: WayPoint::checkName(std::basic_string_view<char, std::char_traits<char> >) const - Gothic2Notr(_ZNK8WayPoint9checkNameESt17basic_string_viewIcSt11char_traitsIcEE+0x9a) [0x56106dec645a]
#6: WayMatrix::findPoint(std::basic_string_view<char, std::char_traits<char> >, bool) const - Gothic2Notr(_ZNK9WayMatrix9findPointESt17basic_string_viewIcSt11char_traitsIcEEb+0xb3) [0x56106dec0e83]
#7: GameScript::ta_min(std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN10GameScript6ta_minESt10shared_ptrIN7phoenix5c_npcEEiiiiiSt17basic_string_viewIcSt11char_traitsIcEE+0x67) [0x56106dcdb9e7]
#8: std::_Function_handler<void (std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >), GameScript::bindExternal<void (GameScript::*)(std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >)>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void (GameScript::*)(std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >))::{lambda((auto:1)...)#1}>::_M_invoke(std::_Any_data const&, std::shared_ptr<phoenix::c_npc>&&, int&&, int&&, int&&, int&&, int&&, std::basic_string_view<char, std::char_traits<char> >&&) - Gothic2Notr(_ZNSt17_Function_handlerIFvSt10shared_ptrIN7phoenix5c_npcEEiiiiiSt17basic_string_viewIcSt11char_traitsIcEEEZN10GameScript12bindExternalIMS9_FvS3_iiiiiS7_EEEvRKNSt7__cxx1112basic_stringIcS6_SaIcEEET_EUlDpT_E_E9_M_invokeERKSt9_Any_dataOS3_OiSS_SS_SS_SS_OS7_+0x90) [0x56106dd04c70]
#9: std::_Function_handler<void (phoenix::vm&), phoenix::vm::register_external<void, std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> > >(std::basic_string_view<char, std::char_traits<char> >, std::function<void (std::shared_ptr<phoenix::c_npc>, int, int, int, int, int, std::basic_string_view<char, std::char_traits<char> >)> const&)::{lambda(phoenix::vm&)#1}>::_M_invoke(std::_Any_data const&, phoenix::vm&) - Gothic2Notr(_ZNSt17_Function_handlerIFvRN7phoenix2vmEEZNS1_17register_externalIvJSt10shared_ptrINS0_5c_npcEEiiiiiSt17basic_string_viewIcSt11char_traitsIcEEEEEvSB_RKSt8functionIFT_DpT0_EEEUlS2_E_E9_M_invokeERKSt9_Any_dataS2_+0x12c) [0x56106dd11b8c]
#10: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0x966) [0x56106df4c846]
#11: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x56106df4ce61]
#12: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0xb28) [0x56106df4ca08]
#13: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x56106df4ce61]
#14: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(+0x2379fa) [0x56106dced9fa]
#15: GameScript::initializeInstanceNpc(std::shared_ptr<phoenix::c_npc> const&, unsigned long) - Gothic2Notr(_ZN10GameScript21initializeInstanceNpcERKSt10shared_ptrIN7phoenix5c_npcEEm+0x3a4) [0x56106dcee444]
#16: Npc::Npc(World&, unsigned long, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN3NpcC1ER5WorldmSt17basic_string_viewIcSt11char_traitsIcEE+0x4e0) [0x56106dea4af0]
#17: WorldObjects::addNpc(unsigned long, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN12WorldObjects6addNpcEmSt17basic_string_viewIcSt11char_traitsIcEE+0x75) [0x56106ded5645]
#18: GameScript::wld_insertnpc(int, std::basic_string_view<char, std::char_traits<char> >) - Gothic2Notr(_ZN10GameScript13wld_insertnpcEiSt17basic_string_viewIcSt11char_traitsIcEE+0x1c) [0x56106dcde33c]
#19: std::_Function_handler<void (phoenix::vm&), phoenix::vm::register_external<void, int, std::basic_string_view<char, std::char_traits<char> > >(std::basic_string_view<char, std::char_traits<char> >, std::function<void (int, std::basic_string_view<char, std::char_traits<char> >)> const&)::{lambda(phoenix::vm&)#1}>::_M_invoke(std::_Any_data const&, phoenix::vm&) - Gothic2Notr(_ZNSt17_Function_handlerIFvRN7phoenix2vmEEZNS1_17register_externalIvJiSt17basic_string_viewIcSt11char_traitsIcEEEEEvS8_RKSt8functionIFT_DpT0_EEEUlS2_E_E9_M_invokeERKSt9_Any_dataS2_+0x62) [0x56106dcffef2]
#20: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0x966) [0x56106df4c846]
#21: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x56106df4ce61]
#22: phoenix::vm::exec() - Gothic2Notr(_ZN7phoenix2vm4execEv+0xb28) [0x56106df4ca08]
#23: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(_ZN7phoenix2vm4callEPKNS_6symbolE+0x31) [0x56106df4ce61]
#24: phoenix::vm::call(phoenix::symbol const*) - Gothic2Notr(+0x2897aa) [0x56106dd3f7aa]
#25: GameSession::initScripts(bool) - Gothic2Notr(_ZN11GameSession11initScriptsEb+0x52b) [0x56106dd3fd2b]
#26: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - Gothic2Notr(_ZN11GameSessionC1ENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x239) [0x56106dd3ffb9]
#27: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - Gothic2Notr(+0x36b1ac) [0x56106de211ac]
#28: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - Gothic2Notr(+0x2afa36) [0x56106dd65a36]
#29: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - /usr/lib/libstdc++.so.6(+0xe1943) [0x7f43634e1943]
#30: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - /usr/lib/libc.so.6(+0x8c9eb) [0x7f436308c9eb]
#31: GameSession::GameSession(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) - /usr/lib/libc.so.6(+0x110dfc) [0x7f4363110dfc]

[1]    15418 IOT instruction (core dumped)  Gothic2Notr -g /home/christopher/Gothic\ II
Gothic2Notr -g /home/christopher/Gothic\ II  5,56s user 1,16s system 65% cpu 10,295 total
```

**Comment by CReimer** [comment]
Forgot the mention the version. It's 16fff0e63df6a00e9801b64c8c72557dbeecb7f7 (1.0.2207)

**Comment by Try** [comment]
Hi, @CReimer and thanks for reporting!

I think that issue here is OOB access to string. Should be fixed in [c8cd3d6](https://github.com/Try/OpenGothic/commit/c8cd3d68cb0b16159118a537008932ec82884496)

**Comment by CReimer** [comment]
Tested. It works now. Thanks
